lib/logstash/outputs/loggly.rb

```diff
@@ -173,7 +176,12 @@ def format_message(event)
   def send_batch(meta_events)
     split_batches(meta_events.compact).each_pair do |k, batch|
       key, tag = *k
-      url = "#{@proto}://#{@host}/bulk/#{key}/tag/#{tag}"
+      if tag != nil
+        url = "#{@proto}://#{@host}/bulk/#{key}/tag/#{tag}"
+      else
+        url = "#{@proto}://#{@host}/bulk/#{key}"
+      end
```

Comment:
Once again, avoid `if !`. `unless...else` is not better, though. So perhaps use something like this instead?

```ruby
if tag.nil?
  url = "#{@proto}://#{@host}/bulk/#{key}"
else
  url = "#{@proto}://#{@host}/bulk/#{key}/tag/#{tag}"
end
```